tests/integration/constraint_test/test_constraints.py

```diff
@@ -0,0 +1,129 @@
+from tests.integration.base import DBTIntegrationTest, use_profile
+from pathlib import Path
+from typing import Dict
+from dbt.contracts.results import RunResult, RunStatus
+
+
+class TestConstraints(DBTIntegrationTest):
+    @property
+    def schema(self):
+        return "constraints"
+
+    @property
+    def models(self):
+        return "models"
+
+    @property
+    def project_config(self):
+        return {
+            "config-version": 2,
+            "models": {
+                "test": {
+                    "+persist_docs": {
+                        "relation": True,
+                        "columns": True,
+                    },
```

Comment:
I'm just wondering if we need to enable them? I'm feeling it's ok to enable automatically if there is such configs in `schema.yml`.
I also feel a bit weird that the config named `persist_docs` enables the constraints.

Reply:
> I'm just wondering if we need to enable them? I'm feeling it's ok to enable automatically if there is such configs in `schema.yml`.

Fair, I think there's good arguments to be made in either direction. From my perspective, `meta` is usually for user's own metadata / annotation purposes, with no _functional_ effect in dbt, until/unless they opt into that via an extension macro/package or custom config.

> I also feel a bit weird that the config named `persist_docs` enables the constraints.

Agreed. Even if the mechanism to add these constraints is very very similar to `persist_docs`, I'd rather see this enabled via another config. Users (and therefore adapter plugins) can define whichever configs they like. All that would require is another macro call within the materialization, right alongside the calls to `persist_docs`:

https://github.com/databricks/dbt-databricks/blob/5d44dc64315a1a906c24b3223b074de4da60460f/dbt/include/databricks/macros/materializations/table.sql#L24-L26

https://github.com/databricks/dbt-databricks/blob/5d44dc64315a1a906c24b3223b074de4da60460f/dbt/include/databricks/macros/materializations/incremental/incremental.sql#L43-L45

Reply:
Make sense! I will use a different config for constraints.

Reply:
We might also need it in `snapshot.sql`? https://github.com/databricks/dbt-databricks/blob/6dd2a8f7fcdafba56d788b42118fb43abe41a59e/dbt/include/databricks/macros/materializations/snapshot.sql#L96-L98